front-end/src/components/Reactionbar/PostReactionBar.tsx

```diff
@@ -0,0 +1,83 @@
+// Copyright 2019-2020 @paritytech/polkassembly authors & contributors
+// This software may be modified and distributed under the terms
+// of the Apache-2.0 license. See the LICENSE file for details.
+
+import React from 'react';
+import styled from '@xstyled/styled-components';
+
+import { reactions } from './reactions';
+import { usePostReactionsQuery } from '../../generated/graphql';
+import ReactionButton, { ReactionButtonProps } from './ReactionButton';
+
+interface Reaction {
+	id: number
+	reaction: string
+}
+
+interface Props {
+	className?: string
+	postId: number
+}
+
+const PostReactionBar = function ({ className, postId }: Props) {
+	const { data, refetch } = usePostReactionsQuery({ variables: { postId } });
+
+	const reactionMap: { [ key: string ]: ReactionButtonProps; } = {};
+
+	reactions.forEach((reaction) => {
+		const reactionButtonProp: ReactionButtonProps = {
+			count: 0,
+			reaction,
```

Comment:
This seems not to be used. It's duplicated since the info is already in the key.

Reply:
But this is needed in ReactionButtonProps so cannot remove

Reply:
`ReactionButtonProps` shouldn't preventing us to use only what we need, and get rid of anything that isn't. I've done it in https://github.com/paritytech/polkassembly/pull/576/commits/2c632b64dce7eacf0f22b462a17133f8868cde0c